src/nnue/nnue_common.h

```diff
@@ -159,6 +159,48 @@ namespace Stockfish::Eval::NNUE {
               write_little_endian<IntType>(stream, values[i]);
   }
 
+  template <typename IntType>
+  inline IntType read_leb_128(std::istream& stream) {
+      static_assert(std::is_signed_v<IntType>, "Not implemented for unsigned types");
+      IntType result = 0, shift = 0;
+      while (true) {
```

Comment:
an infinite loop here is not acceptable, asking for exploits

Reply:
Rewritten, no infinite loop here.